test/integration/pkcs-keygen.int.c

```diff
@@ -438,6 +542,8 @@ int main() {
             test_setup, test_teardown),
         cmocka_unit_test_setup_teardown(test_rsa_keygen_p11tool_templ,
             test_setup, test_teardown),
+        cmocka_unit_test_setup_teardown(test_keygen_wrong_template,
+            test_setup, test_teardown),
```

Comment:
You do the negative test, which is great but you should add at least one path of functional testing. Perhaps ECC since it's faster.

Reply:
Updated the test to also do positive testing.